core/store/migrate/migrations/0227_dynamic_pipeline_runs.sql

```diff
@@ -0,0 +1,34 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TABLE public.job_pipeline_specs (
+    job_id INT NOT NULL,
+    pipeline_spec_id INT NOT NULL,
+    is_primary BOOLEAN NOT NULL DEFAULT FALSE, -- Cannot use `primary` as column name in Postgres since it is a reserved keyword.
+    CONSTRAINT pk_job_pipeline_spec PRIMARY KEY (job_id, pipeline_spec_id),
+    CONSTRAINT fk_job FOREIGN KEY (job_id) REFERENCES public.jobs(id),
+    CONSTRAINT fk_pipeline_spec FOREIGN KEY (pipeline_spec_id) REFERENCES public.pipeline_specs(id)
+);
+
+CREATE UNIQUE INDEX idx_unique_primary_per_job ON public.job_pipeline_spec(job_id) WHERE is_primary;
+
+-- The moment this runs, we only have one job+pipeline_spec combination per job, complying with the unique index.
+INSERT INTO public.job_pipeline_specs (job_id, pipeline_spec_id, is_primary)
+SELECT id, pipeline_spec_id, TRUE
+FROM public.jobs;
+
+
+ALTER TABLE public.jobs DROP COLUMN pipeline_spec_id; -- Do we use CASCADE here? Does it have any relationship with other tables?
```

Comment:
I would hold off on dropping this column until a later version -- might be better to deprecate the field first. I might be overly cautious here (I know that we recommend NOPs take a DB backup, but I don't think they all do this reliably.)

@jmank88 Is this a worthwhile precaution to take or not worth it?

Reply:
bumping @jmank88

Reply:
Hmm I don't like the idea of compromising to make up for a lack of backups :thinking:
It would be neat if we could do that reliably and generally, but I think there would be many difficulty, messy, and impossible cases. Is this case uniquely important?